task/upload.go

```diff
@@ -453,13 +448,12 @@ func complementCatalystPipeline(tctx *TaskContext, assetSpec api.AssetSpec, call
 		return tctx.Progress.TrackReader(rawSourceFile, size, endProgress), nil
 	}
 
+	fullPath := videoFileName(playbackID)
+	assetSpec.Files = append(assetSpec.Files, api.AssetFile{
+		Type: "source_file",
+		Path: toAssetRelativePath(playbackID, fullPath),
+	})
```

Comment:
Could you do this outside this `comlpetementCatalystPipeline` function? The idea for this function, at least originally, was to keep only the features not implemented yet by Catalyst that we needed to complement later. Exactly to facilitate the changes you are making now, knowing exactly the bits that are/were missing from Catalyst. 

So I think it'd make a lot more sense for this logic to be under `processCatalystCallback` instead, not here, since it's not complementing the missing features but only processing the catalyst callback. After the changes you are doing, this function should be completely removed.

WDYT?